src/sinks/elasticsearch.rs

```diff
@@ -64,13 +67,178 @@ pub struct ElasticSearchConfig {
 
     pub aws: Option<RegionOrEndpoint>,
     pub tls: Option<TlsOptions>,
+    // Deprecated, use normal.bulk_action instead
     pub bulk_action: Option<String>,
+    pub normal: Option<NormalConfig>,
+    pub data_stream: Option<DataStreamConfig>,
 }
 
 impl ElasticSearchConfig {
-    fn bulk_action(&self) -> crate::Result<Template> {
-        let value = self.bulk_action.as_deref().unwrap_or("index");
-        Ok(Template::try_from(value).context(BatchActionTemplate)?)
+    fn bulk_action(&self) -> crate::Result<Option<Template>> {
```

Comment:
Not a big deal, but given `bulk_action` at the top-level is deprecated for `normal.bulk_action` we may want to prefer `normal.bulk_action` when both are set.

Reply:
👍 did the same for `index`